packages/core/parcel-bundler/src/scope-hoisting/hoist.js

```diff
@@ -51,6 +52,7 @@ function hasSideEffects(asset, {sideEffects} = asset._package) {
 module.exports = {
   Program: {
     enter(path, asset) {
+      traverse.cache.clearScope();
```

Comment:
Does this have (performance) side effects for every import of `@babel/traverse` ?

Reply:
Possibly, but it’s the only way. I believe Babel’s behavior may actually be correct. I don’t think `scope.crawl` was meant to be called manually. It is called automatically by Babel when setting up a scope for a path, and only re-crawls that scope, not child scopes. So we need to clear the cache so that Babel will automatically crawl child scopes again when we traverse them.

Ideally we wouldn’t need to re-crawl at all, but some Babel plugins are badly behaved and don’t update the scope properly when they mutate the AST.